### ISSUE TITLE
Bug deleting entry on server that already didn't exist

### ISSUE THREAD

**Issue by Lantizia** [marcelklehr/floccus#82]
### Software versions
* Browser: Chrome 65
* Nextcloud: 12.0.5
* Nextcloud Bookmarks app: 0.11.0
* Floccus: 2.0.6

### Steps to reproduce
1. Donate, whether it works entirely or not yet - this is damned fine software (in theory) and needs supporting.
2. On my home PC I copied all my bookmarks (about 2500 of them) into the bookmark folder that floccus uses for syncronisation.
3. Went onto my work PC (with a new install/profile of Chrome) and tried to sync them down.

### Expected outcome
It to sync up properly and down properly?

### Actual outcome
For some reason when it was syncing up it randomly stopped and said it had difficulty deleting entry with id 68 on the server.  I had to go into the SQL of the Nextcloud database into the bookmarks table and make a fake entry 68 just so it could then delete it.  Not sure why, I can provide extra information if I am able to (as I'll likely re-attempt uploading these all up again in the future when all the bugs I'm listing are ironed out).

**Comment by Lantizia** [comment]
Brill I'll keep an eye out for the next release of that.  I need a rebuild of my Nextcloud server anyway, 12.0.5 is getting oooold.

**Comment by 1stone** [comment]
Hi Marcel,

could you give some background why this is a server-side bug?
With the uncertainty when a new bookmark app will be available, do you see a chance to mitigate this problem in floccus?
I'm currently hitting this problem over and over again (syncing with multiple browsers, Chrome & Firefox). Using the suggested workaround becomes quite annoying...

Cheers

**Comment by marcelklehr** [comment]
Hey @1stone 
Sorry for the delay!
There have been multiple bugs that caused this failure, apparently. In the end the server should be able to handle deletions of items that have already been deleted, which is why I labelled this as an upstream issue. But then, why does the client delete the item in the first place? Initially, I thought this only happens when the same item is deleted concurrently on two clients. But the latest patch release fixes a bug that could lead to all sorts of undesirable behavior. Let me know if v2.2.3 makes your life easier :)

**Comment by e-alfred** [comment]
I have the same problem. I added a few bookmarks in folders I didn't want to and I deleted them on one Floccus client. They were removed from the server without a problem. But then they were synced back from another client. If I delete them on the other client and try "sync now" I get multiple errors complaining "Removing a bookmark on the server failed. remoteId=111".

**Comment by marcelklehr** [comment]
It turns out I hadn't completely fixed the issue I mentioned earlier, so... give v2.2.6 a try :)

**Comment by marcelklehr** [comment]
Did the error come up again in the meantime?

**Comment by e-alfred** [comment]
Yes, it works now. Thanks for the awesome work!

**Comment by 1stone** [comment]
Hmm...
after not seeing this problem for several days, I just encountered it again with 2.2.7 on Chrome.
This is the message copy&pasted from the settings window:

```
Removing a bookmark on the server failed. remoteId=1016 Removing a bookmark on the server failed. remoteId=1017 Removing a bookmark on the server failed. remoteId=1014 Removing a bookmark on the server failed. remoteId=1015 Removing a bookmark on the server failed. remoteId=1018 Removing a bookmark on the server failed. remoteId=1019 Removing a bookmark on the server failed. remoteId=1006 Removing a bookmark on the server failed. remoteId=1005 Removing a bookmark on the server failed. remoteId=1003 Removing a bookmark on the server failed. remoteId=1004
```

**Comment by marcelklehr** [comment]
@1stone I realize that those ids are kind of awkward. Any specific actions that you undertook before this happened?

**Comment by 1stone** [comment]
@marcelklehr: Nothing particular. In fact, I've only revived my workstation from hibernation and noticed that the floccus icon showed an "!" annotation. (It probably was there before. I don't see a relation to hibernation here.)
Clicking on it and saw that error message pasted above.

While two other accounts  (one of that on the same server, but with different path) reported ok, I was not able to change the situation for the failed one. Neither clicking sync or reset had any obvious effect.

Finally, I deleted the account completely and recreated it again with the same settings. A full sync started, and everything was fine again after that.

I'll keep a close eye on this and report back with eventually more infos.
Thanks for your support.
